app/services/user_blocks/channel_handler.rb

```diff
@@ -0,0 +1,37 @@
+module UserBlocks
+  class ChannelHandler
+    attr_reader :user_block
+
+    def initialize(user_block)
+      @user_block = user_block
+    end
+
+    def get_potential_chat_channel
+      blocked_user = User.select(:id, :username).find(user_block.blocked_id)
+      blocker = User.select(:id).find(user_block.blocker_id)
+      blocker.chat_channels.find_by("slug LIKE ? AND channel_type = ?", "%#{blocked_user.username}%", "direct")
```

Comment:
A question, why `LIKE` instead of equality?

Reply:
The `chat_channel.slug` is possibly something like `andaizhao/rhymes` or `rhymes/andaizhao`, so equality would only work if I do something like:
```ruby
chat_channels.find_by("(slug = ? OR slug = ?) AND channel_type = ?", "#{blocked_user.username}/#{blocker.username}", "#{blocker.username}/#{blocked_user.username}", "direct")
```

Which seems too verbose to me.

Reply:
As we discussed, `LIKE` causes inaccuracy problems. The second solution described is better.